### PR TITLE
[SPARK-14516][FOLLOWUP] Adding ClusteringEvaluator to examples

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

In SPARK-14516 we have introduced ClusteringEvaluator, but we didn't put any reference in the documentation and the examples were still relying on the sum of squared errors to show a way to evaluate the clustering model.

The PR adds the ClusteringEvaluator in the examples.

## How was this patch tested?

Manual runs of the examples.

### CODE REVIEW

**Comment by mgaido91** [comment]
sorry for pinging you, what do you think about adding `ClusteringEvaluator` to the examples @yanboliang ? Thanks.

**Comment by yanboliang** [comment]
It's good to have this, sorry for late response, I will make a pass tomorrow. Thanks.

File: examples/src/main/java/org/apache/spark/examples/ml/JavaKMeansExample.java [hunk 1]
```diff
@@ -51,9 +52,17 @@ public static void main(String[] args) {
...
-    // Evaluate clustering by computing Within Set Sum of Squared Errors.
-    double WSSSE = model.computeCost(dataset);
-    System.out.println("Within Set Sum of Squared Errors = " + WSSSE);
+    // Make predictions
+    Dataset<Row> predictions = model.transform(dataset);
+
+    // Evaluate clustering by computing Silhouette score
+    ClusteringEvaluator evaluator = new ClusteringEvaluator()
+      .setFeaturesCol("features")
+      .setPredictionCol("prediction")
```
Comment (yanboliang) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaKMeansExample.java]: We use default values here, so it's not necessary to set them explicitly. We should keep examples as simple as possible. Thanks.

**Review by yanboliang — APPROVED** [review]
LGTM except one minor comment. Thanks.

File: examples/src/main/java/org/apache/spark/examples/ml/JavaKMeansExample.java [hunk 2]
```diff
@@ -51,9 +52,14 @@ public static void main(String[] args) {
...
-    double WSSSE = model.computeCost(dataset);
-    System.out.println("Within Set Sum of Squared Errors = " + WSSSE);
+    // Make predictions
+    Dataset<Row> predictions = model.transform(dataset);
+
+    // Evaluate clustering by computing Silhouette score
+    ClusteringEvaluator evaluator = new ClusteringEvaluator();
+
+    double silhouette = evaluator.evaluate(predictions);
+    System.out.println("Silhouette with squared euclidean distance = " + silhouette);
```
Comment (srowen) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaKMeansExample.java]: euclidean -> Euclidean, but not important to change unless you're touching the code again anyway

Comment (mgaido91) [reply examples/src/main/java/org/apache/spark/examples/ml/JavaKMeansExample.java]: Thanks, I don't think I am changing the code again, but I can fix this grammatical error if you want.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**